src/document/code.rs

```diff
@@ -247,12 +251,15 @@ impl Line {
 
                     if !clean {
                         result.push(format!(
```

Comment:
I understand this was not introduced in this MR, but this function basically does
```rust
let mut result = Vec::new();
...
result.push(format!("...", ...));
...
Ok(result.join("\n"))
```
among other things. Using `write!` like
```
let mut result = String::new();
...
writeln!(&mut result, "...", ...).unwrap();
...
Ok(result)
```
should significantly more efficient.

Reply:
Hm, It is not completely equivalent, as now there is a newline at the end of the line...

Reply:
You can do `result.pop()` before returning if that is an issue.

Reply:
Currently trying exactly that, but with no effect on the result. Maybe the problem is elsewhere, introduced in the same commit.

Reply:
Ah, no, my fault... I was playing with the sources already "broken" by a reverse run. Fixed...